src/index.ts

```diff
@@ -1,46 +1,122 @@
-    const loginCookies = await sso.login();
-    console.log(green("✓"), "Authentication complete");
+async function getJobFromCLI(job: Job, message: string, spinner: ora.Ora) {
+    // display UI
+    const jobs = await job.getJobs();
 
-    const browser = await Puppeteer.launch({ args: ["--no-sandbox"] });
-    const page = await browser.newPage();
-    await sso.setCookies(page, loginCookies);
+    spinner.succeed();
+    const prompt = new Select({
+        name: "Job",
+        message,
+        choices: [...jobs.keys()],
+    });
+
+    const jobName = await prompt.run();
+    return jobs.get(jobName);
+}
 
+async function addPosts(
+    isInteractive: boolean,
+    jobID: number,
+    regions: string[],
+    cloneFrom: number
+) {
+    const sso = new SSO();
+    const { browser, page } = await sso.authenticate();
     const job = new Job(page);
-    const jobData: JobInfo = await job.getJobData(jobID);
+    const spinner = ora("Fetching your jobs.").start();
+
+    try {
+        if (isInteractive) {
+            const id = await getJobFromCLI(
+                job,
+                "What job would you like to create job posts for?",
+                spinner
+            );
+            console.log(`Job ID: ${id}`);
+            // TODO add posts
+        } else {
+            if (!jobID) throw Error(`Job ID argument is missing.`);
+            if (!regions) throw Error(`Region parameter is missing.`);
+            const jobData: JobInfo = await job.getJobData(jobID);
 
-    // Process updates for each 'Canonical' job unless a "clone-from" argument is passed
-    await job.clonePost(jobData.posts, regions, cloneFrom);
+            // Process updates for each 'Canonical' job unless a "clone-from" argument is passed
+            await job.clonePost(jobData.posts, regions, cloneFrom);
+            // Mark all newly added job posts as live
+            await job.markAsLive(jobID, jobData.posts);
+        }
+    } catch (error) {
+        spinner.stop();
+        console.log(`${red("x")} ${(<Error>error).message}`);
+    } finally {
+        browser.close();
+    }
+}
 
-    // Mark all newly added job posts as live
-    await job.markAsLive(jobID, jobData.posts);
+async function deletePosts(
+    isInteractive: boolean,
+    jobID: number,
+    regions: string[],
+    similar: number
+) {
+    const sso = new SSO();
+    const { browser, page } = await sso.authenticate();
+    const job = new Job(page);
+    const spinner = ora("Fetching your jobs.").start();
 
-    browser.close();
+    try {
+        if (isInteractive) {
+            const id = await getJobFromCLI(
+                job,
+                "What job would you like to delete job posts from?",
+                spinner
+            );
+            console.log(`Job ID: ${id}`);
+            // TODO delete posts
+        } else {
+            if (!jobID) throw Error(`Job ID argument is missing.`);
+            await job.deletePosts(jobID, regions, similar);
```

Comment:
Same here, update the spinner message status

Reply:
:+1: